### PR TITLE
instrument retrying channels scheduled executor service

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
We had no metrics on number of pending requests

## After this PR
==COMMIT_MSG==
We produce metrics about requests that are going to be retried
==COMMIT_MSG==

## Possible downsides?
This uses tritium to instrument the scheduledExecutor so the metric names won't be the best

### CODE REVIEW

File: dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java [hunk 1]
```diff
@@ -80,6 +83,7 @@
             ClientConfiguration.RetryOnTimeout retryOnTimeout) {
         this(
                 delegate,
+                new DefaultTaggedMetricRegistry(),
```
Comment (carterkozak) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java:88]: I think we need to add this to the constructor and pass the registry from our ClientConfiguration

File: dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java [hunk 2]
```diff
@@ -245,4 +250,10 @@ private static boolean shouldPropagateQos(ClientConfiguration.ServerQoS serverQo
         throw new SafeIllegalStateException(
                 "Encountered unknown propagate QoS configuration", SafeArg.of("serverQoS", serverQoS));
     }
+
+    private static ListeningScheduledExecutorService instrument(
+            ScheduledExecutorService delegate, TaggedMetricRegistry metrics) {
+        return MoreExecutors.listeningDecorator(
+                Tracers.wrap(SCHEDULER_NAME, MetricRegistries.instrument(metrics, delegate, SCHEDULER_NAME)));
```
Comment (carterkozak) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java:259]: nice

File: dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java (hunk 1 above)
Comment (ferozco) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java:88]: Yes we do do that, this constructor is only visible for testing. Let me  make that more clear

Comment (carterkozak) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java:88]: oh doh, I misread, sorry!

**Review by carterkozak — APPROVED** [review]
thanks!

**Comment by ferozco** [comment]
🤖

**Pull request merged**